packages/gestalt/src/HelpButton.js

```diff
@@ -0,0 +1,185 @@
+   * - Optionally provide an `onClick` callback, which is fired when the link is clicked (pressed and released) with a mouse or keyboard. See [OnLinkNavigationProvider](https://gestalt.pinterest.systems/web/utilities/onlinknavigationprovider) to learn more about link navigation.
+   */
+  link?: {|
+    accessibilityLabel: string,
+    href: string,
+    onClick?: ({|
+      event: SyntheticMouseEvent<HTMLAnchorElement> | SyntheticKeyboardEvent<HTMLAnchorElement>,
+      dangerouslyDisableOnNavigation: () => void,
+    |}) => void,
+    text?: string,
+  |},
+  /**
+   * Callback fired when HelpIcon is clicked (pressed and released) with a mouse or keyboard.
+   */
+  onClick?: OnTapType,
+  /**
+   * Informational content that's displayed when the user clicks on HelpButton.
+   */
+  text: string | Element<typeof Text>,
+  /**
+   * Specifies the z-index for HelpButton's tooltip and popover to resolve any layering issues with other elements. See the [zIndex variant](https://gestalt.pinterest.systems/web/helpbutton#With-Z-index) for more details.
+   */
+  zIndex?: Indexable,
+|};
+
+/**
+ * [HelpButton](https://gestalt.pinterest.systems/web/helpbutton) is an affordance that accompanies an element on the screen. It helps describe or provide assistance on how to use the accompanying element.
+ */
+export default function HelpButton({
+  accessibilityPopoverLabel,
+  idealDirection = 'down',
+  link,
+  onClick,
+  text,
+  zIndex,
+}: Props): Node {
+  const tapAreaRef = useRef(null);
+  const [isOpen, setIsOpen] = useState(false);
+  const [isHovered, setHovered] = useState(false);
+  const [isFocused, setFocused] = useState(false);
+  const popoverId = useId();
+  const { accessibilityTooltipMessage, accessibilityIconLabel, linkLabel } =
+    useDefaultLabelContext('HelpButton');
+
+  useEffect(() => {
+    if (isOpen) {
+      tapAreaRef?.current?.focus();
+    }
+  }, [isOpen]);
+
+  const toggleView = () => {
+    setIsOpen((currVal) => !currVal);
+  };
+
+  const onHandleTap = (props) => {
+    toggleView();
+    onClick?.(props);
+  };
+
+  const bgIconColor = isOpen || isHovered || isFocused ? 'default' : 'subtle';
+
+  const tooltipZIndex = zIndex ?? new FixedZIndex(1);
+
+  const zIndexLayer = new CompositeZIndex([new FixedZIndex(tooltipZIndex.index())]);
+
+  const textElement = typeof text === 'string' ? <Text align="center">{text}</Text> : text;
```

Comment:
Look at Toast, SlimBanner, List... https://github.com/pinterest/gestalt/blob/master/packages/gestalt/src/ListText.js 

We override colors etc There's more logic involved in here. 

We need to eventually dry it but for now try to add similar logic.

Reply:
DONE